### PR TITLE
PLAT-10416 : complete missing endpoints

### PR DESCRIPTION
**Pull request by symphony-soufiane**
### Ticket
[PLAT-10416](https://perzoinc.atlassian.net/browse/PLAT-10416)

### Description
Following [PLAT-10122](https://perzoinc.atlassian.net/browse/PLAT-10122), this PR completes implementation of some missing endpoints : 

- [create-user-v2](https://developers.symphony.com/restapi/reference#create-user-v2) 
- [update-user-v2](https://developers.symphony.com/restapi/reference#update-user-v2)
- [list-audit-trail-v1](https://developers.symphony.com/restapi/reference#list-audit-trail-v1)
- [disclaimer](https://developers.symphony.com/restapi/reference#disclaimer)
- [list-disclaimers](https://developers.symphony.com/restapi/reference#list-disclaimers)
- [disclaimer-users](https://developers.symphony.com/restapi/reference#disclaimer-users)

None of these endpoints are OBO-enabled, so no change is made to the OBO interface.
Changes in core/service/user/UserService are made and corresponding unit tests are added.
DisclaimerService is added and exposed in spring-boot-starter module.

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Javadoc added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java [hunk 1]
```diff
@@ -652,6 +655,33 @@ public FollowingListResponse listUsersFollowing(@Nonnull Long userId, @Nonnull C
...
+
+  /**
+   * Updates an existing V2 Users
+   *
+   * @param userId    User Id
+   * @param payload   User's new attributes for update.
+   * @return {@link V2UserDetail} with updated user details.
+   * @see <a href="https://developers.symphony.com/restapi/reference#update-user-v2">Update User v2</a>
+   */
+  @API(status = API.Status.STABLE)
```
Comment (symphony-soufiane) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java:679]: Is this annotation something that should be automatically generated (after tests..)  or should it be manually written in the code ?

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java [hunk 2]
```diff
@@ -652,6 +655,33 @@ public FollowingListResponse listUsersFollowing(@Nonnull Long userId, @Nonnull C
...
 
+  /**
+   * Return a {@link V2UserDetail} with created user details.
+   *
+   * @param payload User's details to create.
+   * @return a {@link V2UserDetail} with created user details.
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-user-v2">Create User v2</a>
+   */
+  @API(status = API.Status.STABLE)
+  public V2UserDetail create(@Nonnull V2UserCreate payload) {
```
Comment (symphony-soufiane) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java:666]: Is it allowed to pass an OpenAPI generated model in arguments here ?  (like https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/bafc4cf0c3e3cdf8cb934a169952fe3876234a89/symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java#L541 )

Comment (symphony-elias) [reply symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java:666]: It is manually added but it is useless here as it is put at class level (line 70)

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java (hunk 1 above)
Comment (symphony-elias) [reply symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java:679]: It is manually added but it is useless here as it is put at class level (line 70). This is only to tell our users the API is stable, internal or experimental.

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java [hunk 3]
```diff
@@ -652,6 +655,33 @@ public FollowingListResponse listUsersFollowing(@Nonnull Long userId, @Nonnull C
     return new PaginatedService<>(api, pagination.getChunkSize(), pagination.getTotalSize()).stream();
   }
 
+  /**
+   * Return a {@link V2UserDetail} with created user details.
```
Comment (symphony-youri) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java:659]: I would say the method creates a new user (i.e returning something is already said by @return)

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java [hunk 4]
```diff
@@ -652,6 +655,33 @@ public FollowingListResponse listUsersFollowing(@Nonnull Long userId, @Nonnull C
...
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-user-v2">Create User v2</a>
+   */
+  @API(status = API.Status.STABLE)
+  public V2UserDetail create(@Nonnull V2UserCreate payload) {
+    return executeAndRetry("createUser",
+        () -> userApi.v2AdminUserCreatePost(authSession.getSessionToken(), payload));
+  }
+
+  /**
+   * Updates an existing V2 Users
```
Comment (symphony-youri) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java:672]: Just "Update an existing user" ?

**Review by symphony-youri — APPROVED** [review]
Just some minor comments for me + removal of @API that is already set at class level pointed out by @symphony-elias

**Comment by symphony-hong** [comment]
I think we should expose the disclaimer service in the spring boot starter as well?

**Review by symphony-hong — APPROVED** [review]
LGTM! 👍

**Review by symphony-elias — APPROVED** [review]
Looks good !

**Review by symphony-youri — APPROVED** [review]

**Review by symphony-thibault — APPROVED** [review]
👌

**Pull request merged**